Upgrading of SQLalchemy to version 1.3.5

frafra opened the pull request:
Fixes #4736
Replaces #4782

frafra commented:
It would be probably good to merge squashing everything into a single commit.

davidread commented on ckan/config/environment.py:
```diff
@@ -292,6 +292,7 @@ def update_config():
 
     # CONFIGURATION OPTIONS HERE (note: all config options will override
     # any Pylons config options)
+    config.setdefault('pool_pre_ping', True)
```
A one-line comment explaining this wouldn't go amiss.

frafra replied:
Sure, it makes sense.

frafra commented on requirements.txt:
```diff
@@ -20,26 +20,26 @@ funcsigs==1.0.2           # via beaker
 pastedeploy==2.0.1        # via pastescript, pylons
 pastescript==2.0.2
-pbr==5.2.0                # via sqlalchemy-migrate
+pbr==5.3.0                # via sqlalchemy-migrate
```
I agree; I will remove the ones that are not related.

frafra commented on ckan/model/meta.py:
```diff
@@ -74,8 +74,8 @@ def before_commit(self, session):
-                         revision_table.c.current == '1')
-                ).values(current='0')
+                         revision_table.c.current == True)
+                ).values(current= False)
```
Right, thanks.

frafra commented:
I removed some whitespaces and empty lines amending older commits, I added a comment and committed only the part of `requirements.txt` related to this specific upgrade. I also squashed the older commits, so we can have a cleaner history and all the commits can pass the tests, without the need of squashing everything.

iamarnavgarg commented:
> I removed some whitespaces and empty lines amending older commits, I added a comment and committed only the part of `requirements.txt` related to this specific upgrade. I also squashed the older commits, so we can have a cleaner history and all the commits can pass the tests.
> 
> No need to squash in my opinion. And thanks to @iamarnavgarg for the work :-)

Thanks @frafra for helping to make the commits merge.

avdata99 commented:
Considering SQLAlchemy has [medium severity vulnerability](https://snyk.io/vuln/SNYK-PYTHON-SQLALCHEMY-590109) is important to mark this to be backported to CKAN 2.8
